controllers/applicationvolumereplication_controller.go

```diff
@@ -1066,6 +1109,85 @@ func (a *AVRInstance) updateAVRStatus() error {
 	return nil
 }
 
+/*
+Description: create a new ManagedClusterView object, or update the existing one with the same name.
+Requires:
+	1) meta: specifies MangedClusterView name and managed cluster search information
+	2) viewscope: once the managed cluster is found, use this information to find the resource.
+		Optional params: Namespace, Resource, Group, Version, Kind. Resource can be used by itself, Kind requires Version
+Returns: ManagedClusterView, error
+*/
+func (r *ApplicationVolumeReplicationReconciler) createOrGetManagedClusterView(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope) (*fndv2.ManagedClusterView, error) {
+	mcv := &fndv2.ManagedClusterView{
+		ObjectMeta: meta,
+		Spec: fndv2.ViewSpec{
+			Scope: viewscope,
+		},
+	}
+
+	err := r.Get(context.TODO(), types.NamespacedName{Name: meta.Name, Namespace: meta.Namespace}, mcv)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			err = r.Create(context.TODO(), mcv)
+		}
+
+		if err != nil {
+			return nil, errorswrapper.Wrap(err, "failed to createOrGetManagedClusterView")
+		}
+	}
+
+	mcv.Spec.Scope = viewscope // set or update ViewScope with query params
+
+	return mcv, nil
+}
+
+/*
+Description: queries a managed cluster for a resource type, and populates a variable with the results.
+Requires:
+	1) meta: information of the new/existing resource; defines which cluster(s) to search
+	2) viewscope: query information for managed cluster resource. Example: resource, name.
+	3) interface: empty variable to populate results into
+Returns: error if encountered (nil if no error occurred). See results on interface object.
+*/
+func (r *ApplicationVolumeReplicationReconciler) getManagedClusterResource(
+	meta metav1.ObjectMeta, viewscope fndv2.ViewScope, resource interface{}) error {
+	// create MCV first
+	mcv, err := r.createOrGetManagedClusterView(meta, viewscope)
+	if err != nil {
+		err = fmt.Errorf("could not get or create ManagedClusterView")
```

Comment:
Probably you should combine this line with line 1160.  Something like:
return errorswrapper.Wrap(err, "could not get or create ManagedClusterView")

in line 1158, you are losing information about the error.